geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -130,15 +131,23 @@ class HBaseIndexAdapter(ds: HBaseDataStore) extends IndexAdapter[HBaseDataStore]
         }
       }
 
-      // wait for the table to come online
-      if (!admin.isTableAvailable(name)) {
-        val timeout = TableAvailabilityTimeout.toDuration.filter(_.isFinite())
-        logger.debug(s"Waiting for table '$table' to become available with " +
-            s"${timeout.map(t => s"a timeout of $t").getOrElse("no timeout")}")
-        val stop = timeout.map(t => System.currentTimeMillis() + t.toMillis)
-        while (!admin.isTableAvailable(name) && stop.forall(_ > System.currentTimeMillis())) {
-          Thread.sleep(1000)
-        }
+      waitForTable(admin, name)
+    }
+  }
+
+  override def renameTable(from: String, to: String): Unit = {
+    WithClose(ds.connection.getAdmin) { admin =>
+      val existing = TableName.valueOf(from)
+      val renamed = TableName.valueOf(to)
+      if (admin.tableExists(existing)) {
+        // renaming in hbase requires creating a snapshot and using that to create the new table
+        val snapshot = StringSerialization.alphaNumericSafeString(UUID.randomUUID().toString)
+        admin.disableTable(existing)
+        admin.snapshot(snapshot, existing)
+        admin.cloneSnapshot(snapshot, renamed)
+        admin.deleteSnapshot(snapshot)
+        admin.deleteTable(existing)
+        waitForTable(admin, renamed)
```

Comment:
I don't know that cloneSnapshot will enable the table when it creates it.

Reply:
I think I tested this but let me do it again just to be sure.

Reply:
I verified it works, at least on my local pseudo-distributed hbase.